third_party/intel/backend/driver.c

```
@@ -137,7 +137,15 @@ static PyObject *loadBinary(PyObject *self, PyObject *args) {
   const size_t binary_size = PyBytes_Size(py_bytes);
 
   uint8_t *binary_ptr = (uint8_t *)PyBytes_AsString(py_bytes);
-  const auto ctx = sycl_device.get_platform().ext_oneapi_get_default_context();
+  auto platform = sycl_device.get_platform();
+  sycl::context ctx;
+  try {
+    ctx = platform.ext_oneapi_get_default_context();
+  } catch (const std::runtime_error &ex) {
+#if _DEBUG
+    std::cout << "ERROR: " << ex.what() << std::endl;
+#endif
```

Looks like a development-friendly code. Will it stay that way?

---

I am not sure I understand your question. In the original code we don't catch an exception that comes from `ext_oneapi_get_default_context`. On Windows it is always thrown but we can safely ignore it (at least it looks so since everything works just fine). We don't have `_DEBUG` defined even in current *WithAsserts configurations so there is no message coming from this code by default.

---

> On Windows it is always thrown but we can safely ignore it (at least it looks so since everything works just fine).

Yes, that's what caught my eye. Without exception handling, it looks like the code was used when writing the code and then debugging (and subsequent cleanup that might have been forgotten).

Maybe at least add a comment, because ignoring the exception doesn't seem obvious?

---

Ok I added a comment that exception is ignored.